make/validate.mk

```diff
@@ -36,7 +36,7 @@ ifeq "$(strip $(VALIDATE_IMAGES))" "1"
 endif
 
 ifneq "$(strip $(NOT_VALIDATE_TABLES))" "1"
-  FAULTY_TABLES := $(shell $(LIBEXEC_DIR)/validate-tables.py $(SRCFILES) 2>&1 | sed -r -e 's,^/([^/: ]+/)*,,' -e 's,.http://docbook.org/ns/docbook.,,' | sed -rn '/^- / !p' | tr "\n" "@")
+  FAULTY_TABLES := $(shell XML_CATALOG_FILES=/etc/xml/catalog $(LIBEXEC_DIR)/validate-tables.py $(PROFILED_MAIN) 2>&1 | sed -r -e 's,^/([^/: ]+/)*,,' -e 's,.http://docbook.org/ns/docbook.,,' | sed -rn '/^- / !p' | tr "\n" "@")
```

Comment:
Using `XML_CATALOG_FILES` like this could potentially dangerous. If someone set its own catalog(s), it won't be taken into account.

To make sure it uses the customized values or falls back onto the main catalog, try this:

```
XML_CATALOG_FILES=${XML_CATALOG_FILES:-/etc/xml/catalog}  ... rest ...
```

However, I'm not sure if make allows such advanced shell syntax.

Reply:
Ah well, as you can see from the CI, this commit does not work at all, even in its current state:

```
Fatal error:
<string>:
    ERROR:Attempt to load network entity http://www.oasis-open.org/docbook/xml/4.5/dbcentx.mod (line 0 column 0)
<string>:
    ERROR:Attempt to load network entity http://www.oasis-open.org/docbook/xml/4.5/dbcentx.mod (line 0 column 0)
[...]
```

Reply:
That's a pity. Hmn, maybe it's a consecutive fault? I've tried a very minimal example with an entity declaration in the internal subset of the DTD:

<details>
   <summary>XML file with entity in internal subset</summary>

```xml
!DOCTYPE article PUBLIC "-//OASIS//DTD DocBook XML V4.5//EN"
                         "http://docbook.org/xml/4.5/docbookx.dtd"
[
   <!ENTITY ent "Entity">
]>
<article>
    <title>A short document with entities</title>
    <para>&ent;</para>
</article>
```
</details>

With the following script, I've tried to parse it:

<details>
   <summary>Python script</summary>

```python
from lxml import etree
import sys

xmlparser = etree.XMLParser(
                            load_dtd=True,
                            resolve_entities=True,
                            huge_tree=True,
                            )
print("sys.argv=", sys.argv)
etree.parse(sys.argv[1])
```
</details>

When I call the above script it just works (meaning, it doesn't create any tracebacks or other errors). Tested it on openSUSE Leap 15.2 and Ubuntu 20.04.

Reply:
JFYI: I've tested the `validate-tables.py` script separately on a Ubuntu 20.04 machine. Just passed the above file and I don't get any tracebacks.

Reply:
The CI seems to install all of it just fine... `docbook docbook-xml docbook-xsl docbook-xsl-ns docbook-xsl-saxon docbook5-xml`

That's enough packages that I am almost confident none was missed.

Reply:
@tomschr I finally ran the script ... and well, it all works, apparently: https://github.com/openSUSE/daps/runs/5057459852?check_suite_focus=true#step:6:1

Reply:
Oh my. I just noticed that all this time, we were trying to fix the wrong Python script. Fock.
The script that is actually provoking the issue is `daps-xmlwellformed`. (I noticed because I created the branch release-321 which omitted all the validate-tables.py stuff and only includes the `daps-xmlwellformed` changes